chirp/drivers/kenwood_d7.py

```diff
@@ -13,332 +13,3404 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-from chirp.drivers.kenwood_live import KenwoodOldLiveRadio, MODES, STEPS, \
-    DUPLEX, LOG, get_tmode
-from chirp import chirp_common, directory, util
+import logging
+import math
+import os
+import threading
+
+from chirp import chirp_common, directory, errors, util
 from chirp.settings import RadioSetting, RadioSettingGroup, \
     RadioSettingValueInteger, RadioSettingValueBoolean, \
-    RadioSettingValueString, RadioSettingValueList, RadioSettings
+    RadioSettingValueString, RadioSettingValueList, RadioSettingValueMap, \
+    RadioSettings, RadioSettingValueFloat
 
+LOG = logging.getLogger(__name__)
 
-@directory.register
-class THD7Radio(KenwoodOldLiveRadio):
-    """Kenwood TH-D7"""
-    MODEL = "TH-D7"
+
+# It would be nice to have this as an array so it's grouped in the right
+# Unfortunately, when we do that, we can't have labels for each line
+class KenwoodD7Position(RadioSettingGroup):
```

Comment:
These aren't usually subclassed like this. It's not really a problem and they generally don't change much, but none of the other drivers do it (that I know of) so it kinda puts you in being-different-land. The usage is pretty straightforward, so I don't anticipate problems, but it's also one of those things where we won't know until someone tries to use it because it's a live driver. So just FYI...

Reply:
Ditto

Reply:
Oh, this is about the RadioSetting subclasses, not the comment... :smile: 

Yeah, this seemed like the cleanest way to bundle all the code together and keep the _SETTINGS dict as the only thing the individual drivers needed.  I've actually made the settings "stuff" a mixin that I'll be using for a TS-2000 overhaul I've hacked up that I'm very happy with (which is also much faster, and has all the menu settings).  I'll try to get as many live radios using it as possible so it's noticed quickly. :wink: